### ISSUE TITLE
Square brackets in URLs cause vale to lint within the URL (markdown)

### ISSUE THREAD

**Issue by Ravlen** [vale-cli/vale#278]
@jdkato A minor bug we just ran into. If the URL has square brackets (either `[` or `]` seem to trigger the bug), Vale will lint within the "code" when it shouldn't:

- <img width="1038" alt="Screen Shot 2020-12-02 at 11 40 30" src="https://user-images.githubusercontent.com/36979740/100821425-b10a2100-3493-11eb-87bc-3d0ab8f21e7b.png">
- <img width="1038" alt="Screen Shot 2020-12-02 at 11 40 51" src="https://user-images.githubusercontent.com/36979740/100821438-ba938900-3493-11eb-8769-ae48bd245184.png">

**Comment by jdkato** [comment]
I can't reproduce this.

Is the file that this happens on open source?

**Comment by Ravlen** [comment]
@jdkato Yes, and I also have some more details. On Friday a co-worker was able to clear these false positives by excluding all "quad backtick" codeblocks from linting: https://gitlab.com/gitlab-org/gitlab/-/merge_requests/49154/diffs#5bce789523695fa2be00cc10575b9cf889c61fa4_1929_1942

I'm not sure if this is related to https://github.com/errata-ai/vale/issues/271 or not, though.

I dug into that file, and deleted as much as I could to see what was the culprit. I got it down to the following, which still causes false positives with multiple rules (both `scope:raw` and regular):

`````
Currently a blacklist [link](https://gitlab.com/name[]).

````markdown
```ruby
external_url "https://gitlab.example.com"
```

```yaml
gitlab:
  host: "gitlab.example.com"
```
````
`````

<img width="837" alt="Screen Shot 2020-12-07 at 10 12 55" src="https://user-images.githubusercontent.com/36979740/101299307-19327b80-3875-11eb-9bf8-54111521458d.png">

It turns out the false positive I'm highlighting in the screenshot above is actually from the codeblock, but the square brackets are causing it to be reported in the URL (I think?). If I delete the square brackets, the warning moves down to the codeblock:

<img width="837" alt="Screen Shot 2020-12-07 at 10 16 51" src="https://user-images.githubusercontent.com/36979740/101299375-5860cc80-3875-11eb-9e32-8fee17b06fcb.png">

Aha, if you delete the nested triple backtick codeblocks, all false positives disappear:

`````
Currently a blacklist [link](https://gitlab.com/name[]).

````markdown
external_url "https://gitlab.example.com"

gitlab:
  host: "gitlab.example.com"
````
`````

<img width="837" alt="Screen Shot 2020-12-07 at 10 17 47" src="https://user-images.githubusercontent.com/36979740/101299454-a1188580-3875-11eb-9cee-bb4352ab5b2e.png">

If you delete the quad backtick code blocks, they disappear too:

````
Currently a blacklist [link](https://gitlab.com/name).

```ruby
external_url "https://gitlab.example.com"
```

```yaml
gitlab:
  host: "gitlab.example.com"
```
````

<img width="837" alt="Screen Shot 2020-12-07 at 10 20 29" src="https://user-images.githubusercontent.com/36979740/101299526-db822280-3875-11eb-84bf-3f83956f3599.png">

Finally, if you put another layer of single backticks, the text seems to be ignored again:

`````
Currently a blacklist [link](https://gitlab.com/name).

````markdown
```ruby
external_url "https://gitlab.example.com"
```
````

````markdown
```ruby
`external_url "https://gitlab.example.com"`
```
````
`````

<img width="837" alt="Screen Shot 2020-12-07 at 10 23 24" src="https://user-images.githubusercontent.com/36979740/101299647-44699a80-3876-11eb-914e-395eff4190f8.png">

**Comment by jdkato** [comment]
Thanks for the extra info. I think there were two issues at play here and both should be fixed in the v2.6.5.
